src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -1010,6 +1033,27 @@ void IrcMessageHandler::handleUserNoticeMessage(Communi::IrcMessage *message,
         {
             messageText = "Announcement";
         }
+        else if (msgType == "subgift" &&
+                 ANONYMOUS_GIFTER_ID == tags.value("user-id").toString())
+        {
+            if (auto monthsIt = tags.find("msg-param-gift-months");
+                monthsIt != tags.end())
+            {
+                int months = monthsIt.value().toInt();
+                if (months > 1)
+                {
+                    messageText =
+                        QString("An anonymous user gifted %1 months of a Tier "
+                                "%2 sub to %3!")
+                            .arg(QString::number(months),
+                                 tags.value("msg-param-sub-plan")
+                                     .toString()
+                                     .at(0),
```

Comment:
This can throw if `msg-param-sub-plan` is set, but is an empty string.
See how SubtierPredicate.cpp does it https://github.com/Chatterino/chatterino2/blob/master/src/messages/search/SubtierPredicate.cpp#L25-L26